cpp/src/plasma/store.h

```diff
@@ -203,42 +165,49 @@ class PlasmaStore {
   /// Remove all of the GetRequests for a given client.
   ///
   /// @param client The client whose GetRequests should be removed.
-  void RemoveGetRequestsForClient(Client* client);
+  void RemoveGetRequestsForClient(const std::shared_ptr<PlasmaClientConnection>& client);
 
   void ReturnFromGet(GetRequest* get_req);
 
   void UpdateObjectGetRequests(const ObjectID& object_id);
 
-  int RemoveFromClientObjectIds(const ObjectID& object_id, ObjectTableEntry* entry,
-                                Client* client);
-
   uint8_t* AllocateMemory(size_t size, int* fd, int64_t* map_size, ptrdiff_t* offset);
+
+  void IncreaseObjectRefCount(const ObjectID& object_id, ObjectTableEntry* entry);
+
+  void DecreaseObjectRefCount(const ObjectID& object_id, ObjectTableEntry* entry);
+
 #ifdef PLASMA_CUDA
   Status AllocateCudaMemory(int device_num, int64_t size, uint8_t** out_pointer,
                             std::shared_ptr<CudaIpcMemHandle>* out_ipc_handle);
 #endif
 
-  /// Event loop of the plasma store.
-  EventLoop* loop_;
+  /// Accept a client connection.
+  void DoAccept();
+  /// Handle an accepted client connection.
+  void HandleAccept(const asio::error_code& error);
```

Comment:
Better put a Status here as a return code. We should try to minimize the asio code outside of the io module.

Reply:
It is specified by the asio acceptor. How about using `std::error_code`? `asio::error_code` is just an alias of `std::error_code` with C++11. I think I just forget to convert it here.